AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift

```diff
@@ -616,6 +616,90 @@ public class ConstructorIO: CIOSessionManagerDelegate {
+        - customerID: The item ID.
+        - variationID: The variation ID
+        - itemName: The product item name
+        - resultID: The identifier of result set returned by the Constructor quiz response
+        - resultPage: The current page of the results
+        - resultCount: The total number of results
+        - numResultsPerPage: The number of results on the current page
+        - resultPositionOnPage: The position of clicked item
+        - sectionName The name of the autocomplete section the result came from
+        - completionHandler: The callback to execute on completion.
+     
+     ### Usage Example: ###
+     ```
+     constructorIO.trackQuizResultClick(quizID: 'coffee-quiz', quizVersionID: '1231244', quizSessionID: '123', customerID: '123', itemName: 'espresso')
+     ```
+     */
+    public func trackQuizResultClick(quizID: String, quizVersionID: String, quizSessionID: String, customerID: String, variationID: String? = nil, itemName: String? = nil, resultID: String? = nil, resultPage: Int? = nil, resultCount: Int? = nil, numResultsPerPage: Int? = nil, resultPositionOnPage: Int? = nil, sectionName: String? = nil, completionHandler: TrackingCompletionHandler? = nil) {
+        let section = sectionName ?? self.config.defaultItemSectionName ?? Constants.Track.defaultItemSectionName
+        let data = CIOTrackQuizResultClickData(quizID: quizID, quizVersionID: quizVersionID, quizSessionID: quizSessionID, customerID: customerID, variationID: variationID, itemName: itemName, resultID: resultID, resultPage: resultPage, resultCount: resultCount, numResultsPerPage: numResultsPerPage, resultPositionOnPage: resultPositionOnPage, sectionName: section)
+        let request = self.buildRequest(data: data)
+        executeTracking(request, completionHandler: completionHandler)
+    }
+
+    /**
+     Track when a user clicks a result that appears within a quizzes results page
+
+     - Parameters:
+        - quizID: The quiz identifier
+        - quizVersionID: The quiz version identifier
+        - quizSessionID: The quiz session identifier associated with this conversion event
+        - customerID: The item ID.
+        - variationID: The variation ID
+        - itemName: The product item name
+        - revenue: The sale price if available, otherwise the regular (retail) price of item
+        - conversionType: The type of conversion (defaults to "add_to_cart")
+        - isCustomType: The flag to specify if type is custom conversion type
+        - displayName: The display name for the custom conversion type
+        - sectionName The name of the autocomplete section the result came from
+        - completionHandler: The callback to execute on completion.
+ 
+     ### Usage Example: ###
+     ```
+     constructorIO.trackQuizConversion(quizId: 'coffee-quiz', quizVersionId: '1231244', quizSessionId: '3123', itemName: 'espresso', variationId: '167', type: 'add_to_cart', revenue: 1.0)
```

Comment:
I think `type` should be `conversionType` here right? I think the order of parameters might be slightly off as well in the example compared to the function signature.

We could probably remove it since it defaults to `add_to_cart` though.

Reply:
Good catch! I'll remove it since `add_to_cart` is the default